### ISSUE TITLE
IntelliJ console is not detected from a code started via Gradle's JavaExec task

### ISSUE THREAD

**Issue by jonnyzzz** [ajalt/mordant#7]
I have a Java application with the library and start it from IntelliJ IDEA via a Gradle JavaExec task. 
The console is not detected in that scenario. 

The detector code actually fails on the line
>   if (!consoleAvailable()) return NONE

**Comment by jonnyzzz** [comment]
The actual environment variables from the Gradle session under IntelliJ (committing obvious ones):
```
JAVA_MAIN_CLASS_27354=com.jonnyzzz.teamcity.rr.RViewMain
LANG=en_US.UTF-8
LESS=-R
LSCOLORS=Gxfxcxdxbxegedabagacad
OLDPWD=/
PAGER=less
SHELL=/bin/zsh
TERM=xterm
TOOLBOX_VERSION=1.18.7455
USER=jonnyzzz
VERSIONER_PYTHON_VERSION=2.7
XPC_FLAGS=0x0
XPC_SERVICE_NAME=com.jetbrains.intellij.29860.373ECF8D-4625-4FA3-B1D7-91E0EE789D61
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
```

The `TERM=xterm` is actually set

**Comment by ajalt** [comment]
Checking `System.console() != null` is all we have access to on the JVM. If you know of a way to improve detection here outside of JNI, I'd love to hear it. But unfortunately, I'm not sure there's anything we can do.

If you know you're going to be running on an interactive terminal, you can force a given color level when creating your `TermColors` instance. For example, to always output truecolor:

```kotlin
TermColors(TermColors.Level.TRUECOLOR)
```

**Comment by jonnyzzz** [comment]
It looks like the TERM variable check can be made before the check of the
console to be available. It seems like (but please tell me if I'm wrong)
that coloring options are available for that case and the caret moves are
not.

**Comment by ajalt** [comment]
I don't think that's reliable. Ignoring `System.console` would result in a false-positive if the environment variable exists but stdout isn't interactive (e.g. you pipe the output of a command somewhere).

**Comment by jonnyzzz** [comment]
I agree. It would be nice allow manually call the terminal mode detection without the console mode check. That could be a solution to my case, so I could call the detection and ignore the console mode check. Would that make sense?

**Comment by ajalt** [comment]
Yes, that seems like a good idea. I'll look into implementing the option to force interactivity while still detecting other terminal info.

**Comment by ajalt** [comment]
I implemented this suggestion in 2.0.0-alpha2: If you create your terminal with `Terminal(interactive = true)`, it will skip the console check, but will still look at environment variables to determine ansi support.
